src/vcl/http.rs

```diff
@@ -263,7 +259,7 @@ fn header_from_hd<'a>(txt: *const ffi::txt) -> Option<(&'a str, &'a str)> {
         let e = (*txt).e;
         let buf = from_raw_parts(b.cast::<u8>(), e.offset_from(b) as usize);
 
-        let colon = buf.iter().position(|x| *x == b':').unwrap();
+        let colon = buf.iter().position(|x| *x == b':')?;
```

Comment:
that one is an actual invariant, the lower levels guarantee it

Reply:
sure, but do we actually want to add exception throwing code here? I think it is easier to just return None if for some magical reason it has it?

Reply:
no, if that invariant isn't maintained, something is very wrong, possibly because some memory corruption, we should blow up early and loudly so we (`varnishd` really) can fix the issue. Hiding the problem will hurt us in the long run

Reply:
thanks, makes sense, fixed